src/test/groovy/com/github/maleksandrowicz93/cqrsdemo/student/StudentApiSpec.groovy

```diff
@@ -0,0 +1,291 @@
+package com.github.maleksandrowicz93.cqrsdemo.student
+
+
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.ErrorMessage
+import com.google.gson.Gson
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.web.servlet.MockMvc
+import spock.lang.Specification
+
+import static org.hamcrest.Matchers.hasSize
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@SpringBootTest
+@ContextConfiguration(classes = IntegrationTestConfig.class)
+@AutoConfigureMockMvc
+class StudentApiSpec extends Specification {
+
+    @Autowired
+    MockMvc mockMvc
+    @Autowired
+    Gson gson
+    @Autowired
+    StudentRepository studentRepository
+
+    def setup() {
+        studentRepository.deleteAll()
+    }
+
+    def "get all students"() {
+        given: "2 students exist in db"
+        def student = studentRepository.save(StudentUtils.studentToAdd())
+        def student1 = studentRepository.save(StudentUtils.alternativeStudentToAdd())
```

Comment:
nie jestem przekonany co do tych nazw zmiennych - łatwo o literówkę i to `1` budzi rozważanie, dlaczego ten drugi to nie `0` xD

Reply:
good point! do tego, utrudnia refactoring nazwy, bo rzutuje na commenty i stringi :)